clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -447,7 +505,8 @@ private ConsumerGroupHeartbeatRequest getHeartbeatRequest(HeartbeatRequestManage
     @ParameterizedTest
     @MethodSource("errorProvider")
     public void testHeartbeatResponseOnErrorHandling(final Errors error, final boolean isFatal) {
-        mockStableMember();
+        if (isFatal)
+            when(membershipManager.state()).thenReturn(MemberState.FATAL);
```

Comment:
trying to understand why exactly we need this for, I ended up in the `ensureHeartbeatStopped` that asserts on the `membershipMgr` state to check it's fatal, was that the reason? I would say we should remove that assert on the mock that brings no value (and then probably this, if not required for anything else)

Reply:
Looks like the if statement was not necessary, the tests pass regardless. Removing the assert and stubbing still make the test work!

Reply:
great, that's what I was expecting.